src/features/app-interfaces/components/create-app-interface-form.tsx

```diff
@@ -50,10 +66,14 @@ export function CreateAppInterfaceForm({ appSpecFile, appSpec, onSuccess }: Prop
   const defaultValues = useMemo(
     () => ({
       file: appSpecFile,
-      name: snapshot.context.name ?? appSpec.contract.name,
+      name: isArc32AppSpec(appSpec)
+        ? snapshot.context.name ?? appSpec.contract.name
+        : isArc4AppSpec(appSpec)
+          ? snapshot.context.name ?? appSpec.name
+          : '',
```

Comment:
Nitpick: you don't need to repeat `snapshot.context.name`, this can be 
```
      name: snapshot.context.name ?? (isArc32AppSpec(appSpec) ? appSpec.contract.name : isArc4AppSpec(appSpec) ? appSpec.name : ''),

```